From de6fa4d6b97fe6d8e26bc58257ef7568f5408098 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Tue, 01 Oct 2019 17:19:29 +0300
Subject: [PATCH] [SW-0] gaudi: remove w/a - first stage

In the first stage, we remove the w/a from all PLLs, but PCI.
We still leave 2 w/a:
1) PCI PLL is still configured in BTL - output 0 and 1 are switched in
   BTL, output 2 and 3 are switched in the W/A callback in preboot.
2) Preboot location is still in scratch pad SRAM.

Change-Id: I8d31901ddf3b14b38209851597c4d5dc2c887bd7
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/gaudi/asic.c b/gaudi/asic.c
index 49ff186..dea8115 100644
--- a/gaudi/asic.c
+++ b/gaudi/asic.c
@@ -26,8 +26,8 @@
 			     HL_PLL_REF_CLK,
 			     HL_PLL_REF_CLK
 		},
-		.div_wa = { HL_PLL_WA_INIT,
-			    HL_PLL_WA_INIT,
+		.div_wa = { HL_PLL_WA_NOWA,
+			    HL_PLL_WA_NOWA,
 			    HL_PLL_WA_NOWA,
 			    HL_PLL_WA_NOWA
 		},
@@ -36,11 +36,11 @@
 		.nr = 0, .nf = 63, .od = 1, .nb = 31, /* 1600 */
 		.div_factor = { 0, 1, 2, 7 },
 		.div_sel = { HL_PLL_PLL_CLK,
-			     HL_PLL_REF_CLK,
-			     HL_PLL_REF_CLK,
-			     HL_PLL_REF_CLK
+			     HL_PLL_DIV_PLL,
+			     HL_PLL_DIV_PLL,
+			     HL_PLL_DIV_PLL
 		},
-		.div_wa = { HL_PLL_WA_INIT,
+		.div_wa = { HL_PLL_WA_NOWA,
 			    HL_PLL_WA_NOWA,
 			    HL_PLL_WA_NOWA,
 			    HL_PLL_WA_NOWA
@@ -50,11 +50,11 @@
 		.nr = 0, .nf = 63, .od = 1, .nb = 31, /* 1600 */
 		.div_factor = { 0, 1, 2, 7 },
 		.div_sel = { HL_PLL_PLL_CLK,
-			     HL_PLL_REF_CLK,
-			     HL_PLL_REF_CLK,
-			     HL_PLL_REF_CLK
+			     HL_PLL_DIV_PLL,
+			     HL_PLL_DIV_PLL,
+			     HL_PLL_DIV_PLL
 		},
-		.div_wa = { HL_PLL_WA_INIT,
+		.div_wa = { HL_PLL_WA_NOWA,
 			    HL_PLL_WA_NOWA,
 			    HL_PLL_WA_NOWA,
 			    HL_PLL_WA_NOWA
@@ -68,8 +68,8 @@
 			     HL_PLL_REF_CLK,
 			     HL_PLL_REF_CLK
 		},
-		.div_wa = { HL_PLL_WA_INIT,
-			    HL_PLL_WA_INIT,
+		.div_wa = { HL_PLL_WA_NOWA,
+			    HL_PLL_WA_NOWA,
 			    HL_PLL_WA_NOWA,
 			    HL_PLL_WA_NOWA
 		},
@@ -78,11 +78,11 @@
 		.nr = 0, .nf = 59, .od = 1, .nb = 29, /* 1500 */
 		.div_factor = { 0, 1, 2, 7 },
 		.div_sel = { HL_PLL_PLL_CLK,
-			     HL_PLL_REF_CLK,
-			     HL_PLL_REF_CLK,
-			     HL_PLL_REF_CLK
+			     HL_PLL_DIV_PLL,
+			     HL_PLL_DIV_PLL,
+			     HL_PLL_DIV_PLL
 		},
-		.div_wa = { HL_PLL_WA_INIT,
+		.div_wa = { HL_PLL_WA_NOWA,
 			    HL_PLL_WA_NOWA,
 			    HL_PLL_WA_NOWA,
 			    HL_PLL_WA_NOWA
@@ -96,7 +96,7 @@
 			     HL_PLL_REF_CLK,
 			     HL_PLL_REF_CLK
 		},
-		.div_wa = { HL_PLL_WA_INIT,
+		.div_wa = { HL_PLL_WA_NOWA,
 			    HL_PLL_WA_NOWA,
 			    HL_PLL_WA_NOWA,
 			    HL_PLL_WA_NOWA
@@ -106,11 +106,11 @@
 		.nr = 0, .nf = 59, .od = 1, .nb = 29, /* 1500 */
 		.div_factor = { 0, 1, 2, 7 },
 		.div_sel = { HL_PLL_PLL_CLK,
-			     HL_PLL_REF_CLK,
-			     HL_PLL_REF_CLK,
-			     HL_PLL_REF_CLK
+			     HL_PLL_DIV_PLL,
+			     HL_PLL_DIV_PLL,
+			     HL_PLL_DIV_PLL
 		},
-		.div_wa = { HL_PLL_WA_INIT,
+		.div_wa = { HL_PLL_WA_NOWA,
 			    HL_PLL_WA_NOWA,
 			    HL_PLL_WA_NOWA,
 			    HL_PLL_WA_NOWA
@@ -124,7 +124,7 @@
 			     HL_PLL_REF_CLK,
 			     HL_PLL_REF_CLK
 		},
-		.div_wa = { HL_PLL_WA_INIT,
+		.div_wa = { HL_PLL_WA_NOWA,
 			    HL_PLL_WA_NOWA,
 			    HL_PLL_WA_NOWA,
 			    HL_PLL_WA_NOWA
@@ -138,9 +138,9 @@
 			     HL_PLL_DIV_PLL,
 			     HL_PLL_DIV_PLL
 		},
-		.div_wa = { HL_PLL_WA_INIT,
-			    HL_PLL_WA_INIT,
-			    HL_PLL_WA_INIT,
+		.div_wa = { HL_PLL_WA_NOWA,
+			    HL_PLL_WA_NOWA,
+			    HL_PLL_WA_NOWA,
 			    HL_PLL_WA_NOWA
 		},
 	}, {
@@ -167,9 +167,9 @@
 			     HL_PLL_DIV_PLL,
 			     HL_PLL_DIV_PLL
 		},
-		.div_wa = { HL_PLL_WA_INIT,
-			    HL_PLL_WA_INIT,
-			    HL_PLL_WA_INIT,
+		.div_wa = { HL_PLL_WA_NOWA,
+			    HL_PLL_WA_NOWA,
+			    HL_PLL_WA_NOWA,
 			    HL_PLL_WA_NOWA
 		},
 	}, {
@@ -181,8 +181,8 @@
 			     HL_PLL_REF_CLK,
 			     HL_PLL_REF_CLK
 		},
-		.div_wa = { HL_PLL_WA_INIT,
-			    HL_PLL_WA_INIT,
+		.div_wa = { HL_PLL_WA_NOWA,
+			    HL_PLL_WA_NOWA,
 			    HL_PLL_WA_NOWA,
 			    HL_PLL_WA_NOWA
 		},
@@ -195,7 +195,7 @@
 			     HL_PLL_DIV_PLL,
 			     HL_PLL_REF_CLK
 		},
-		.div_wa = { HL_PLL_WA_INIT,
+		.div_wa = { HL_PLL_WA_NOWA,
 			    HL_PLL_WA_NOWA,
 			    HL_PLL_WA_NOWA,
 			    HL_PLL_WA_NOWA
@@ -209,7 +209,7 @@
 			     HL_PLL_DIV_PLL,
 			     HL_PLL_REF_CLK
 		},
-		.div_wa = { HL_PLL_WA_INIT,
+		.div_wa = { HL_PLL_WA_NOWA,
 			    HL_PLL_WA_NOWA,
 			    HL_PLL_WA_NOWA,
 			    HL_PLL_WA_NOWA
@@ -237,9 +237,9 @@
 			     HL_PLL_DIV_PLL,
 			     HL_PLL_DIV_PLL
 		},
-		.div_wa = { HL_PLL_WA_INIT,
-			    HL_PLL_WA_INIT,
-			    HL_PLL_WA_INIT,
+		.div_wa = { HL_PLL_WA_NOWA,
+			    HL_PLL_WA_NOWA,
+			    HL_PLL_WA_NOWA,
 			    HL_PLL_WA_NOWA
 		},
 	}, {
@@ -251,9 +251,9 @@
 			     HL_PLL_DIV_PLL,
 			     HL_PLL_DIV_PLL
 		},
-		.div_wa = { HL_PLL_WA_INIT,
-			    HL_PLL_WA_INIT,
-			    HL_PLL_WA_INIT,
+		.div_wa = { HL_PLL_WA_NOWA,
+			    HL_PLL_WA_NOWA,
+			    HL_PLL_WA_NOWA,
 			    HL_PLL_WA_NOWA
 		},
 	},
@@ -276,6 +276,7 @@
 	case mmPSOC_CPU_PLL_BASE:
 		return false;
 	case mmNIC2_HBM_PLL_BASE:
+		break;
 		if (gaudi_hbw_init_done(cfg->base) && gaudi_reset_soc_done())
 			return false;
 		break;
@@ -408,6 +409,7 @@
 		case mmPSOC_CPU_PLL_BASE:
 			continue;
 		case mmNIC2_HBM_PLL_BASE:
+			continue;
 			if (gaudi_reset_soc_done()) {
 				regmap_writel(0, mmPSOC_GLOBAL_CONF_COLD_RST_FLOPS_2);
 				/* reinitialize the PCIe on next reset */
@@ -422,14 +424,12 @@
 				gaudi_reset_soc();
 			}
 			break;
+		case mmPSOC_PCI_PLL_BASE:
+			hl_pll_wa_clk_sw(&gaudi_pll_cfg[i]);
+			hl_uart_init();
+			break;
 		default:
 			break;
 		}
-
-		hl_pll_wa_clk_sw(&gaudi_pll_cfg[i]);
-
-		/* if we've just switched the UART clock, reinitialize it */
-		if (gaudi_pll_cfg[i].base == mmPSOC_PCI_PLL_BASE)
-			hl_uart_init();
 	}
 }
